flux_led/__main__.py

```diff
@@ -574,32 +578,84 @@ def _determineMode(self, ww_level, pattern_code):
                 mode = "color"
         elif pattern_code == 0x60:
             mode = "custom"
+        elif pattern_code == 0x41:
+            mode = "color"
         elif PresetPattern.valid(pattern_code):
             mode = "preset"
         elif BuiltInTimer.valid(pattern_code):
             mode = BuiltInTimer.valtostr(pattern_code)
-
         return mode
 
-    def update_state(self, retry=2):
+ 
+    def _determine_query_len(self):
+
+        # determine the type of protocol based of first 2 bytes.
+        self._use_csum = True
+        self._send_msg(bytearray([0x81, 0x8a, 0x8b]))
+        rx = self._read_msg(2)
```

Comment:
i edited the code.

now, it will try to determine the query length 3 times before exiting. If no response is recieved, _send_msg and _read_msg still execute, but as the expected len is 0, they immediately exit.